pkg/forwarder/transactions_file_storage.go

```diff
@@ -162,11 +183,13 @@ func (s *transactionsFileStorage) reloadExistingRetryFiles() error {
 	sort.Slice(files, func(i, j int) bool {
 		return files[i].ModTime().Before(files[j].ModTime())
 	})
-
+	var filenames []string
 	for _, file := range files {
 		fullPath := path.Join(s.storagePath, file.Name())
-		s.filenames = append(s.filenames, fullPath)
+		filenames = append(filenames, fullPath)
 	}
+	s.telemetry.addReloadedRetryFilesCount(len(filenames))
+	s.filenames = filenames
```

Comment:
With this change the behavior is changed here (filenames is not growing the same way). Is the new one the one you want?

Reply:
`reloadExistingRetryFiles` is called in `newTransactionsFileStorage` and so `s.filenames` is nil and the behavior is the same. 
I will updated the code to make it more explicit.

Reply:
Code improved.